### PR TITLE
Elide pytest-internal paths for --fixtures

### LINKED ISSUE #8822: Investigate cleaning up of --fixture path outputs
those look broken

_Originally posted by @RonnyPfannschmidt in https://github.com/pytest-dev/pytest/pull/8814#discussion_r660076919_

### PR DESCRIPTION
**Pull request by The-Compiler**
Before this PR:

```
caplog -- .tox/py39-pyqt515/lib/python3.9/site-packages/_pytest/logging.py:476
    Access and control log capturing
```

with it:

```
caplog -- .../_pytest/logging.py:491
    Access and control log capturing.
```

Note that we still display the full path for fixtures from plugins, though. I think a fix for that would be a bit more involved, and probably not something for 7.0.

Fixes #8822

### CODE REVIEW

**Comment by The-Compiler** [comment]
cc  @rahul-kumi if you want to take a look!

File: src/_pytest/python.py [hunk 1]
```diff
@@ -1550,14 +1563,13 @@ def _showfixtures_main(config: Config, session: Session) -> None:
...
-        tw.write(f" -- {bestrel}", yellow=True)
+        tw.write(f" -- {prettypath}", yellow=True)
         tw.write("\n")
-        loc = getlocation(fixturedef.func, str(curdir))
         doc = inspect.getdoc(fixturedef.func)
         if doc:
             write_docstring(tw, doc.split("\n\n")[0] if verbose <= 0 else doc)
         else:
-            tw.line(f"    {loc}: no docstring available", red=True)
+            tw.line(f"    no docstring available", red=True)
```
Comment (The-Compiler) [inline src/_pytest/python.py:1572]: removed the location here as it seems redundant now that we always show the location anyways.

**Review by nicoddemus — APPROVED** [review]
Good job!

File: src/_pytest/python.py [hunk 2]
```diff
@@ -1443,6 +1446,16 @@ def idmaker(
     return resolved_ids
 
 
+def _pretty_fixture_path(func) -> str:
+    cwd = Path.cwd()
+    loc = Path(getlocation(func, str(cwd)))
+    prefix = Path("...", "_pytest")
+    try:
+        return str(prefix / loc.relative_to(_PYTEST_DIR))
```
Comment (RonnyPfannschmidt) [inline src/_pytest/python.py:1454]: How about we opt for relative to site packages if in site packages and/or relative to start dir if sensible

I believe the real bug is that bestrelpath fails to determine when the absolute path is actually shorter than the best relative one

**Review by RonnyPfannschmidt — APPROVED** [review]

File: src/_pytest/python.py (hunk 2 above)
Comment (The-Compiler) [reply src/_pytest/python.py:1454]: Do we have something already to check if a path is in site packages? From what I remember from things like `hunter`, getting a proper site packages path is quite a pain with differences between Python versions, distributions (Debian/Ubuntu and `dist-packages`), etc. etc.

I suppose bestrelpath could show an absolute path if that is shorter indeed. But given that it's used in various places I'm not sure if I want to introduce such a change...

Comment (The-Compiler) [reply src/_pytest/python.py:1454]: Oh: The path from my example above *is* relative and "correct", but I still believe the elided path is nicer for the output.

Comment (RonnyPfannschmidt) [reply src/_pytest/python.py:1454]: The proposed solution is already a enhancement, let's take my comment as a follow-up

Comment (The-Compiler) [reply src/_pytest/python.py:1454]: Opened #9292 for the follow-up.

**Pull request merged**